router/docs/Configuration.md

```diff
@@ -0,0 +1,101 @@
+# Router Configuration Conventions
+
+This document outlines the conventions used in Cosmo's configuration schema.
+
+## General Structure
+
+- Configuration is defined in a JSON Schema format
+- All configuration options have specific types with validation constraints
+- Properties use `snake_case` naming convention
+- Nested objects are used to group related configuration options
+
+## Validation Rules
+
+- Required fields are explicitly marked with `required` arrays
+- `additionalProperties: false` ensures no unknown properties are allowed
+- Minimum/maximum constraints are specified where applicable
+- Enums are used for properties with a fixed set of allowed values
+
+## Common Patterns
+
+- Property names use nouns or configurations, **not** verb-prefixed names (use `storage` not `get_storage` or `use_storage`)
+- Every configurable component uses an `enabled` boolean property to toggle features without removing them from the configuration
+- URLs follow the format `scheme://host:port`
+- File paths use the `format: file-path` validator
+- Sizes are specified using string format with units (e.g., "100MB")
+- Time durations use string format with units (e.g., "1s", "1m", "1h")
+- Default values are specified in `pkg/config/config.go` via struct tags.
```

Comment:
Perhaps we should write something about preferring boolean config values to default to `false`? i.e. `IncludeSchemaUsage` > `ExcludeSchemaUsage`

Reply:
Good point. Feel free to add it.